### PR TITLE
Add coreutils timeout tool

### PR DESCRIPTION
**Pull request by mcgov**
Timeout tool for scheduling a command with timeout. Optional parameter to send a kill signal after the first signal.

### CODE REVIEW

File: lisa/tools/timeout.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    @property
+    def command(self) -> str:
+        return "timeout"
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def run_with_timeout(
+        self, command: str, signal: str, timeout: int, kill_timeout: int = 0
```
Comment (squirrelsc) [inline lisa/tools/timeout.py]: Give the signal a default value.

**Comment by squirrelsc** [comment]
I'm thinking about how the tool can be integrated with existing tools deeper. But so far it can be merged to use firstly. Current tools have the timeout underlying, but the timeout is controlled by force kill. We may consider supporting signal for graceful kill.

**Comment by mcgov** [comment]
I have a use for it coming in the next PR :)

**Comment by squirrelsc** [comment]
> 

ok, add a default value for signal, which can gracefully kill.

**Comment by mcgov** [comment]
Refactored the SIG constants to util.constants now that they're shared between Kill and Timeout. Set default signal to SIGTERM

File: lisa/tools/timeout.py [hunk 2]
```diff
@@ -11,7 +12,7 @@ def can_install(self) -> bool:
         return False
 
     def run_with_timeout(
-        self, command: str, signal: str, timeout: int, kill_timeout: int = 0
+        self, command: str, timeout: int, signal: int = SIGTERM, kill_timeout: int = 0
```
Comment (squirrelsc) [inline lisa/tools/timeout.py:15]: How about put SIGINT as default? Is it softer?

Comment (mcgov) [reply lisa/tools/timeout.py:15]: I think TERM is the 'nice' way to ask for a kill normally. Testpmd likes INT in interactive mode but it's less common (I think).

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**